libc/calls/write-nt.c

```diff
@@ -52,6 +52,11 @@ static textwindows ssize_t sys_write_nt_impl(int fd, void *data, size_t size,
   struct Fd *f = g_fds.p + fd;
   bool isconsole = f->kind == kFdConsole;
 
+  // not implemented
+  if (f->kind == kFdDevRandom) {
+    return enosys();
```

Comment:
I think you can just say `eperm()` here, that's what I get on a XNU system when I try to write to `/dev/urandom`.

Reply:
For me on Linux:
```bash
echo dfdff > /dev/urandom
echo $?
0
```

`man 4 random` on my Linux system says:

> Writing to /dev/random or /dev/urandom will  update  the  entropy  pool
       with  the  data  written,  but this will not result in a higher entropy
       count.  This means that it will impact  the  contents  read  from  both
       files, but it will not make reads from /dev/random faster.

Does it work with root?

I figured `enosys()` was fitting as updating the entropy pool is not possible or at least not implemented, but if there's already a convention on returning `eperm()` we could use it instead.

Reply:
Wow, wacky. I dunno if you should be allowed to drive after smoking whatever those guys are on.

No it doesn't work with root on XNU.

But this unfortunately makes the tests more annoying to write.

I guess I'd say use `eperm()` and keep the write tests `IsWindows()`-gated.